src/ZODB/DB.py

```diff
@@ -645,6 +645,15 @@ def _(c):
         self._mvcc_storage.close()
         del self.storage
         del self._mvcc_storage
+        # clean up references to other DBs
+        self.databases.clear()
```

Comment:
In fact, I just looked at some code that clearing the dict would break.

@agroszer just reset the attr as @jamadden suggested.